meilisearch/src/routes/multi_search.rs

```diff
@@ -46,7 +46,7 @@ pub async fn multi_search_with_post(
     // Explicitly expect a `(ResponseError, usize)` for the error type rather than `ResponseError` only,
     // so that `?` doesn't work if it doesn't use `with_index`, ensuring that it is not forgotten in case of code
     // changes.
-    let search_results: Result<_, (ResponseError, usize)> = (|| {
+    let search_results: Result<_, (ResponseError, usize)> = {
```

Comment:
Looks OK, I think the initial implemetnation wasn't async, and required the immediately executed lambda so that `?` inside of the then-non-async block wouldn't return from the enclosing function.

Since the value is async now, the `?` already doesn't return from the enclosing function, but from the async block itself, which is OK.

Actually it can be further reduced from:

```rust
{
    async {
    // ...
    }
}.await
```

to just
```rust
async { /*...*/}.await
```